src/common/state/error_table.h

```diff
@@ -0,0 +1,60 @@
+#ifndef ERROR_TABLE_H
+#define ERROR_TABLE_H
+
+#include "db.h"
+#include "table.h"
+
+/* Data structures that are needed for pushing error messages to Redis. These
+ * data structures are only used internally. */
+
+typedef struct {
+  /* The ID of the driver that the error should be pushed to. */
+  unique_id driver_id;
+  /* The string used as an identifier for this error message. */
+  unique_id error_id;
+  /* The length of the error type. */
+  int64_t error_type_len;
+  /* The length of the error message. */
+  int64_t error_message_len;
+  /* The concatenation of the error type and error message. The size of this
+   * array will actually be strlen(error_type) + strlen(error_message). The
+   * error type will come first, followed by the error message. The substrings
+   * in this array are not null-terminated. */
+  uint8_t error_type_and_message[1];
```

Comment:
Hmm. I don't think that really works, since we don't get to control how the data is freed, right?

That's handled automatically by `table.c`, which wouldn't know to call `free` on the callback_data fields.